contrib/udfs/src/test/java/org/apache/drill/exec/udfs/TestNearestDateFunctions.java

```diff
@@ -0,0 +1,158 @@
+            "nearestDate( '2019-02-15 07:48:00', 'yyyy-MM-dd HH:mm:ss', 'QUARTER_HOUR') AS nearest_quarter_hour, " +
+            "nearestDate( '2019-02-15 07:22:00', 'yyyy-MM-dd HH:mm:ss', 'MINUTE') AS nearest_minute, " +
+            "nearestDate( '2019-02-15 07:22:22', 'yyyy-MM-dd HH:mm:ss', 'HALF_MINUTE') AS nearest_30second, " +
+            "nearestDate( '2019-02-15 07:22:22', 'yyyy-MM-dd HH:mm:ss', 'QUARTER_MINUTE') AS nearest_15second, " +
+            "nearestDate( '2019-02-15 07:22:31', 'yyyy-MM-dd HH:mm:ss', 'SECOND') AS nearest_second " +
+            "FROM (VALUES(1))";
+    testBuilder()
+            .sqlQuery(query)
+            .unOrdered()
+            .baselineColumns("nearest_year",
+                    "nearest_quarter",
+                    "nearest_month",
+                    "nearest_day",
+                    "nearest_week_sunday",
+                    "nearest_week_monday",
+                    "nearest_hour",
+                    "nearest_half_hour",
+                    "nearest_quarter_hour",
+                    "nearest_minute",
+                    "nearest_30second",
+                    "nearest_15second",
+                    "nearest_second")
+            .baselineValues(LocalDateTime.of(2019, 1, 1, 0, 0, 0),  //Year
+                    LocalDateTime.of(2019, 1, 1, 0, 0, 0),  //Quarter
+                    LocalDateTime.of(2019, 2, 1, 0, 0, 0), //Month
+                    LocalDateTime.of(2019, 2, 15, 0, 0, 0), //Day
+                    LocalDateTime.of(2019, 2, 10, 0, 0, 0), //Week Sunday
+                    LocalDateTime.of(2019, 2, 11, 0, 0, 0), //Week Monday
+                    LocalDateTime.of(2019, 2, 15, 7, 0, 0), //Hour
+                    LocalDateTime.of(2019, 2, 15, 7, 30, 0), //Half Hour
+                    LocalDateTime.of(2019, 2, 15, 7, 45, 0), //Quarter Hour
+                    LocalDateTime.of(2019, 2, 15, 7, 22, 0), //Minute
+                    LocalDateTime.of(2019, 2, 15, 7, 22, 0), //30Second
+                    LocalDateTime.of(2019, 2, 15, 7, 22, 15), //15Second
+                    LocalDateTime.of(2019, 2, 15, 7, 22, 31)) //Second
+            .go();
+  }
+
+  @Test
+  public void testInvalidDateInterval() throws Exception {
+    String query = "SELECT nearestDate( TO_TIMESTAMP('2019-02-01 07:22:00', 'yyyy-MM-dd HH:mm:ss'), 'BAD_DATE') AS nearest_year " +
+            "FROM (VALUES(1))";
+    try {
+      testBuilder()
+              .sqlQuery(query)
+              .expectsEmptyResultSet()
+              .baselineColumns("nearest_year")
+              .go();
+    } catch (DrillRuntimeException e) {
+      final String pattern = "\\[BAD_DATE\\] is not a valid time statement\\.".toLowerCase();
+      final boolean badDateFound = e.getMessage().toLowerCase().contains(pattern);
+      System.err.println("Error Message: " + e.getMessage());
+      assertTrue(badDateFound);
+      throw e;
```

Comment:
```suggestion
```

Reply:
Hi @arina-ielchiieva 
I made the requested changes, but the unit test now fails with the following errors:

```
6:14:46.585 [2375d669-997a-33ee-e9de-28b6048db2a6:foreman] ERROR o.a.drill.exec.work.foreman.Foreman - SYSTEM ERROR: DrillRuntimeException: [BAD_DATE] is not a valid time statement. Expecting: [YEAR, QUARTER, MONTH, WEEK_SUNDAY, WEEK_MONDAY, DAY, HOUR, HALF_HOUR, QUARTER_HOUR, MINUTE, HALF_MINUTE, QUARTER_MINUTE, SECOND]


Please, refer to logs for more information.

[Error Id: 41b715ec-a71a-4e28-8441-fe74be92d26c on 192.168.1.21:31013]
org.apache.drill.common.exceptions.UserException: SYSTEM ERROR: DrillRuntimeException: [BAD_DATE] is not a valid time statement. Expecting: [YEAR, QUARTER, MONTH, WEEK_SUNDAY, WEEK_MONDAY, DAY, HOUR, HALF_HOUR, QUARTER_HOUR, MINUTE, HALF_MINUTE, QUARTER_MINUTE, SECOND]


Please, refer to logs for more information.
```

Any suggestions?

Reply:
I guess this is obvious, just change DrillUserException to UserException....

Reply:
Unfortunately, that didn't fix it either.  I tried generic `Exception`, `RpcException` etc.

Reply:
Not sure what it is/was but I fixed the unit test in question.